pkg/testutils/leakcheck.go

```diff
@@ -28,6 +28,14 @@ func IgnoreGlogFlushDaemonLeak() goleak.Option {
 	return goleak.IgnoreTopFunction("github.com/golang/glog.(*fileSink).flushDaemon")
 }
 
+// IgnoreElasticHealthCheckerLeak returns a goleak.Option that ignores the healthchecker function
+// from the olivere/elastic package that can cause false positives in leak detection.
+// This is necessary because elastic starts a goroutine in the background that may not
+// be stopped when the test finishes, leading to a detected but expected leak.
+func IgnoreElasticHealthCheckerLeak() goleak.Option {
+	return goleak.IgnoreTopFunction("github.com/olivere/elastic.(*Client).healthchecker")
+}
```

Comment:
no, we don't want to do that. We ignore the other ones because those libs always leak goroutines with no ability to shut them down. In contrast, this ES client leaking is our fault, the unit test is kinda unstable and doesn't properly close all clients.

Reply:
Can we look a way of using defer Close() method?

Reply:
Yes, we need to track down the leaks and ensure Close are called as needed.

Reply:
> I have tried different methods of using close methods

trying random things is not how to solve this problem. We need to identify which specific test is causing the leak, and understand what causes the ES client to leak.